### PR TITLE
Add function `SchemaType.prototype.validateAll`

### LINKED ISSUE #6910: Add validateMany feature method
**Do you want to request a *feature* or report a *bug*?**
Feature

**What is the current behavior?**
There is no way to run multiple validators on the shame field.

```
toySchema.path('name').validate(function(v) {
  if (v !== 'Turbo Man') {
    throw new Error('Need to get a Turbo Man for Christmas');
  }
  return true;
}, 'Name `{VALUE}` is not valid');
```

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**
We should to be able to add multiple custom validators, and get back the result all of them.

I saw that there were some issue what reported the same, and the answer was, we can use mongoose-validator-all package. But that package have no more maintained for 1,5 years and anyway it should be supported by mongoose i think.

To prevent breaking changes, instead of overwrite the current validate system, just add a new one. 
```
toySchema.path('password').validateMany([
function(v, f) {
  if (this.name == v) {    
   return {
        'passed': false,
         'message': `field [${f}] can be the same as name, value: [${v}]`
    };
  }
  return {
    'passed': true,
  };
},function(v, f) {
  if (this.confirmPw == v) {    //just for example
   return {
        'passed': false,
         'message': `field [${f}] have to be the same as password, confirmPw, password: [${this.confirmPw}, ${v}]`
    };
  }
  return {
    'passed': true,
  };
},
]);
```

Errors:
```
{
 "password": [{error Object}, {error Object}]
}
```

UI:
For be able to compare fields whit others e.g password, this should to contain the document itself as in pre hooks.

### PR DESCRIPTION
**Pull request by IslandRhythms**
closes #6910

### CODE REVIEW

File: lib/schemaType.js [hunk 1]
```diff
@@ -803,6 +803,12 @@ SchemaType.prototype.get = function(fn) {
   return this;
 };
 
+SchemaType.prototype.validateAll = function(approvalSeekers) {
```
Comment (hasezoey) [inline lib/schemaType.js:806]: consider adding JSDoc.

File: test/schematype.test.js [hunk 2]
```diff
@@ -281,4 +281,29 @@ describe('schematype', function() {
       });
     });
   });
+  it('demonstrates the `validateAll()` function (gh-6910)', async function() {
+
+    const validateSchema = new Schema({ name: String, password: String });
```
Comment (hasezoey) [inline test/schematype.test.js:286]:
```suggestion
  it('demonstrates the `validateAll()` function (gh-6910)', async function() {
    const validateSchema = new Schema({ name: String, password: String });
```

File: test/schematype.test.js [hunk 3]
```diff
@@ -281,4 +281,29 @@ describe('schematype', function() {
       });
     });
   });
+  it('demonstrates the `validateAll()` function (gh-6910)', async function() {
```
Comment (hasezoey) [inline test/schematype.test.js:284]:
```suggestion
  it('demonstrates the `validateAll()` function (gh-6910)', function() {
```

function does not need to be async, from what i can tell

**Review by hasezoey — COMMENTED** [review]
Looks good, would like to see some JSDoc and Typescript types.
Also some minor style nitpicks.

PS: changed the title to better reflect what this issue is about

**Review by hasezoey — APPROVED** [review]

**Review by vkarpov15 — APPROVED** [review]

**Pull request merged**